### ISSUE TITLE
Bug: [@typescript-eslint/no-extra-semi] Rule not working with type

### ISSUE THREAD

**Issue by szulcus** [typescript-eslint/typescript-eslint#7274]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&fileType=.tsx&code=C4TwDgpgBAKhDOwoF4oHJgOGg3AWACgBLAO0wCcAzAQwGNoARCCMORKAb0Kik0QC5YWfAQC%2BOIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge3oA99oBDWskQBbSujABtcNhwB3ftCaQZEALoyAviE1A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
type Test = 'test';
interface DeepTest {
	test: Test;
};
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-extra-semi": ["warn"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Show warning to Test type.

### Actual Result

Shows warning only to interface.

### Additional Info

![image](https://github.com/typescript-eslint/typescript-eslint/assets/48328739/2cd2bb34-963e-4c9f-9437-0d8da41db1b0)

**Comment by bradzacher** [comment]
As [mentioned in our documentation for `semi`](https://typescript-eslint.io/rules/semi/#examples) there is a separate rule for this case - https://typescript-eslint.io/rules/member-delimiter-style

**Comment by szulcus** [comment]
@bradzacher From what I understood, the `@typescript-eslint/member-delimiter-style` rule is only for separating object properties. I should be on a warning if the type has a semicolon at the end of the line.

The `@typescript-eslint/semi` rule works, but is the opposite of that. In addition, it also works for standard variables in JS (let and const).

My intention is to require semicolon in JS variables and disallow semicolon in TS types:
```ts
// GOOD
type Test = 'test'
const test = 'test';
// BAD
type Test = 'test';
const test = 'test'
```
Is it possible?

**Comment by bradzacher** [comment]
Oh sorry.

`no-extra-semi` warns against semis in unnecessary places. It warns at the end of the interface because semi isn't syntactically allowed at the end of an interface - so that semi isn't actually part of the interface.

However at the end of a type alias declaration it is syntactically allowed and included in the type alias decl as a termination of the type "expression".

If you're looking to have a no-semi style, then the `semi` rule is what you want.
